### PR TITLE
1457 initially theorems are indefinitely pending

### LINKED ISSUE #1457: initially, theorems are indefinitely pending
after uploading an ontology with an (unproven) theorem, the theorem indefinitely has status "pending", see https://ontohub.org/sandbox/open2//sp///theorems

### PR DESCRIPTION
**Pull request by eugenk**
This shall fix #1457 and additionally slightly improve the state html-widget.

### CODE REVIEW

**Comment by tillmo** [comment]
There is a problem in one of the tests (see travis):
```
Failure/Error: expect(response.body).

     JSON::Schema::ValidationError:

       The property '#/evaluation_state' value "not_started_yet" did not match one of the following values: done, failed, pending, processing
```

**Comment by eugenk** [comment]
With the pull requests of ontohub/ontohub-api-json (referenced above), this should be done now.

**Comment by eugenk** [comment]
Now, it's finally done. There were problems with the API JSON schema tests. I needed to sort out a few problems in the tests here and in the JSON schema files in the above-mentioned pull requests.

File: spec/factories/theorem_factory.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
     name { generate :sentence_name }
     text { Faker::Lorem.sentence }
     proof_status { create :proof_status_open }
-    state { 'pending' }
+    state { 'not_started_yet' }
```
Comment (eugenk) [inline spec/factories/theorem_factory.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

File: spec/controllers/api/v1/symbols_controller_spec.rb [hunk 2]
```diff
@@ -8,7 +8,7 @@
   context 'on GET to index' do
     context 'requesting json representation', api_specification: true do
       let(:symbols_schema) do
-        schema_for('ontology/commands/symbols')
+        schema_for_command('ontology/symbols', :get, 200)
```
Comment (eugenk) [inline spec/controllers/api/v1/symbols_controller_spec.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

File: spec/controllers/api/v1/sentences_controller_spec.rb [hunk 3]
```diff
@@ -8,7 +8,7 @@
   context 'on GET to index' do
     context 'requesting json representation', api_specification: true do
       let(:sentences_schema) do
-        schema_for('ontology/commands/sentences')
+        schema_for_command('ontology/sentences', :get, 200)
```
Comment (eugenk) [inline spec/controllers/api/v1/sentences_controller_spec.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

File: spec/controllers/api/v1/ontology_versions_controller_spec.rb [hunk 4]
```diff
@@ -8,7 +8,7 @@
   context 'on GET to index' do
     context 'requesting json representation', api_specification: true do
       let(:ontology_versions_schema) do
-        schema_for('ontology/commands/ontology_versions')
+        schema_for_command('ontology/ontology_versions', :get, 200)
```
Comment (eugenk) [inline spec/controllers/api/v1/ontology_versions_controller_spec.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

File: spec/controllers/api/v1/children_controller_spec.rb [hunk 5]
```diff
@@ -7,7 +7,7 @@
   context 'on GET to index' do
     context 'requesting json representation', api_specification: true do
       let(:children_schema) do
-        schema_for('ontology/commands/children')
+        schema_for_command('ontology/children', :get, 200)
```
Comment (eugenk) [inline spec/controllers/api/v1/children_controller_spec.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

File: spec/controllers/api/v1/mappings_controller_spec.rb [hunk 6]
```diff
@@ -8,7 +8,7 @@
   context 'on GET to index' do
     context 'requesting json representation', api_specification: true do
       let(:mappings_schema) do
-        schema_for('ontology/commands/mappings')
+        schema_for_command('ontology/mappings', :get, 200)
```
Comment (eugenk) [inline spec/controllers/api/v1/mappings_controller_spec.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

File: lib/state.rb [hunk 7]
```diff
@@ -1,14 +1,17 @@
...
-     done: 'label-success',
-     failed: 'label-danger',
-     }
+    not_started_yet: 'label-default',
+    failed: 'label-danger',
+    no_result: 'label-warning',
+    pending: 'label-warning',
+    fetching: 'label-primary',
+    processing: 'label-primary',
+    done: 'label-success',
```
Comment (eugenk) [inline lib/state.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

File: lib/state.rb [hunk 8]
```diff
@@ -1,14 +1,17 @@
...
-     processing: 'label-primary',
-     done: 'label-success',
-     failed: 'label-danger',
-     }
+    not_started_yet: 'label-default',
+    failed: 'label-danger',
+    no_result: 'label-warning',
+    pending: 'label-warning',
+    fetching: 'label-primary',
+    processing: 'label-primary',
```
Comment (eugenk) [inline lib/state.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

File: lib/state.rb [hunk 9]
```diff
@@ -1,14 +1,17 @@
...
-     pending: 'label-warning',
-     fetching: 'label-primary',
-     processing: 'label-primary',
-     done: 'label-success',
-     failed: 'label-danger',
-     }
+    not_started_yet: 'label-default',
+    failed: 'label-danger',
+    no_result: 'label-warning',
+    pending: 'label-warning',
```
Comment (eugenk) [inline lib/state.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

File: lib/state.rb [hunk 10]
```diff
@@ -1,14 +1,17 @@
...
+    fetching: 'label-primary',
```
Comment (eugenk) [inline lib/state.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

File: lib/state.rb [hunk 11]
```diff
@@ -1,14 +1,17 @@
...
   STATE_LABEL = {
-     pending: 'label-warning',
-     fetching: 'label-primary',
-     processing: 'label-primary',
-     done: 'label-success',
-     failed: 'label-danger',
-     }
+    not_started_yet: 'label-default',
+    failed: 'label-danger',
+    no_result: 'label-warning',
```
Comment (eugenk) [inline lib/state.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

File: lib/state.rb [hunk 12]
```diff
@@ -1,14 +1,17 @@
...
 
   STATE_LABEL = {
-     pending: 'label-warning',
-     fetching: 'label-primary',
-     processing: 'label-primary',
-     done: 'label-success',
-     failed: 'label-danger',
-     }
+    not_started_yet: 'label-default',
+    failed: 'label-danger',
```
Comment (eugenk) [inline lib/state.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

File: lib/state.rb [hunk 13]
```diff
@@ -1,14 +1,17 @@
...
+  IDLE_STATES = TERMINAL_STATES + %w(not_started_yet)
 
   STATE_LABEL = {
-     pending: 'label-warning',
-     fetching: 'label-primary',
-     processing: 'label-primary',
-     done: 'label-success',
-     failed: 'label-danger',
-     }
+    not_started_yet: 'label-default',
```
Comment (eugenk) [inline lib/state.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

File: app/models/ontology/sentences.rb [hunk 14]
```diff
@@ -155,7 +155,7 @@ def set_theorem_attributes(theorem, hash)
             theorem.state = 'done'
           else
             status_id = ProofStatus::DEFAULT_OPEN_STATUS
-            theorem.state = 'pending'
+            theorem.state = 'not_started_yet'
```
Comment (eugenk) [inline app/models/ontology/sentences.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

File: features/support/env.rb [hunk 15]
```diff
@@ -17,6 +17,7 @@
 class Cucumber::Rails::World
   require Rails.root.join('spec', 'support', 'common_helper_methods.rb')
   require Rails.root.join('spec', 'support', 'scenario_progress_formatter.rb')
+  require Rails.root.join('spec', 'support', 'shared_helper.rb')
```
Comment (eugenk) [inline features/support/env.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

File: app/controllers/api/v1/combinations_controller.rb [hunk 16]
```diff
@@ -15,7 +15,7 @@ def create
     else
       response.status = 400
       render json: {status: response.message,
-                    error: errors_str}
+                    errors: [errors_str]}
```
Comment (eugenk) [inline app/controllers/api/v1/combinations_controller.rb]: No thanks. You should stop suggesting this as soon as #1574 gets merged.

**Comment by eugenk** [comment]
Please review https://github.com/ontohub/ontohub-api-json/pull/2 and https://github.com/ontohub/ontohub-api-json/pull/3 as well.

**Pull request merged**